--- linux-2.6.24/Makefile	2008-01-30 00:51:05.836052238 +0200
+++ linux-2.6.24/Makefile	2008-01-30 00:49:23.827629619 +0200
@@ -207,6 +207,23 @@
 
 KCONFIG_CONFIG	?= .config
 
+# PLD Linux pykconfig addon
+pykconfig: .config.conf
+	@echo 'pykconfig is up to date'
+
+.config.conf: $(srctree)/arch/$(SRCARCH)/defconfig
+	@echo '  kernel-config-update.py $(ARCH) arch/$(SRCARCH)/defconfig.conf $< > $@'
+	$(Q)$(srctree)/scripts/kernel-config-update.py $(ARCH) $(srctree)/arch/$(SRCARCH)/defconfig.conf $< > .config.conf.tmp
+	$(Q)mv .config.conf.tmp $@
+
+$(srctree)/arch/$(SRCARCH)/defconfig: $(srctree)/arch/$(SRCARCH)/defconfig.conf
+	@echo '  kernel-config.py $(ARCH) arch/$(SRCARCH)/defconfig.conf arch/$(SRCARCH)/defconfig'
+	$(Q)> .defconfig.tmp
+	$(Q)$(srctree)/scripts/kernel-config.py $(ARCH) $(srctree)/arch/$(SRCARCH)/defconfig.conf .defconfig.tmp
+	$(Q)mv .defconfig.tmp $(srctree)/arch/$(SRCARCH)/defconfig
+	@echo '  ln -sf arch/$(SRCARCH)/defconfig .config'
+	$(Q)ln -sf $(srctree)/arch/$(SRCARCH)/defconfig .config
+
 # SHELL used by kbuild
 CONFIG_SHELL := $(shell if [ -x "$$BASH" ]; then echo $$BASH; \
 	  else if [ -x /bin/bash ]; then echo /bin/bash; \
